src/auth/GetWindowsCert.ts

```diff
@@ -0,0 +1,32 @@
+import * as ca from "win-ca";
+import * as caApi from "win-ca/api";
+import * as crypto from "crypto";
+
+export const UseWindowsCerts = (id: string): string => {
+  const thumbprint = (cert: string) => {
+    var shasum = crypto.createHash("sha1");
+    shasum.update(Buffer.from(cert, "base64"));
+    return shasum.digest("hex").toUpperCase();
+  };
+
+  const list: string[] = [];
+  let returnValue: string = "Certificate Not Found";
+
+  caApi({
+    store: ["My"],
+    ondata: list,
+  });
+
+  list.forEach((cert) => {
+    const certThumbprint = thumbprint(cert);
+    if (certThumbprint === id) {
+      const toPEM = ca.der2(ca.der2.pem);
+      const pem = toPEM(cert);
+      return pem;
+    }
+  });
+
+  return returnValue;
```

Comment:
If no cert was found, the current logic will return the string "Certificate Not Found", which will be used as a cert and cause the auth to fail. If No cert was found, we should instead fail the auth process already without issuing unnecessary requests to AAD knowing they will fail anyway.

Reply:
What would be the preferred fail handler?

Reply:
We should return an error message saying "The specified certificate was not found in the certificate store"

Reply:
added a throw exception if cert not found, which is then caught by the main cert method. That would output the error without trying to connect. 
Not sure of this logic and error handler process so would like your thoughts.